src/net_processing.cpp

```diff
@@ -2438,9 +2441,27 @@ void ProcessMessage(
         return;
     }
 
-    if (msg_type == NetMsgType::ADDR) {
+    if (msg_type == NetMsgType::ADDR || msg_type == NetMsgType::ADDRv2) {
+        const auto version_orig = vRecv.GetVersion();
+        if (msg_type == NetMsgType::ADDRv2) {
+            vRecv.SetVersion(version_orig | ADDRv2_FORMAT);
```

Comment:
I have never seen this variable used like this before and I've just spent 10 minutes trying to figure out what's going on, and I'm still a little confused. Maybe worth adding a comment?
I think this code is counter-intuitive atm.

Reply:
Yes, that is a bit hackish but I do not think there is a cleaner way. It is somewhat similar to [iostream::setf()](http://www.cplusplus.com/reference/ios/ios_base/setf/) which is used to configure the stream wrt how data should be parsed or printed, e.g.:
```cpp
cout.setf(ios::boolalpha);
cout << true; // prints (serializes as) "true" rather than "1"
```

I added comments like:
```cpp
// Sneak ADDRv2_FORMAT into the version so that the CNetAddr and CAddress
// unserialize methods know that an address in v2 format is coming.
```
to all places where we do that.